### PR TITLE
Add option to hide searchissue

### PR DESCRIPTION
**Pull request by stonebuzz**
### Changes description

New option to display or not search Issue (juste toggle the switch)

image

image

### Checklist

Please check if your PR fulfills the following specifications:

- [ ] Tests for the changes have been added
- [ ] Docs have been added/updated

### References

Closes #N/A

### CODE REVIEW

File: inc/entityconfig.class.php [hunk 1]
```diff
@@ -315,6 +326,22 @@ public function showFormForEntity(Entity $entity) {
       }
       echo '</td></tr>';
 
+      // Search issue visibility
+      $elements = self::getEnumSearchIssueVisibility();
+      if ($entityId == 0) {
+         unset($elements[self::CONFIG_PARENT]);
+      }
+      echo "<tr class='tab_bg_1'>";
+      echo "<td>".__('Search issue')."</td>";
```
Comment (btry) [inline inc/entityconfig.class.php:335]:
```suggestion
      echo "<td>".__('Search issue', 'formcreator')."</td>";
```

**Review by btry — CHANGES_REQUESTED** [review]
A suggestion + need to have same defaults for install and upgrade.

We need to consider to have this feature avaialble in GLPI core, as I don't like much to use JS to hide things in the UI.

**Comment by stonebuzz** [comment]
> We need to consider to have this feature avaialble in GLPI core, as I don't like much to use JS to hide things in the UI.

done =)

File: install/upgrade_to_2.13.php [hunk 2]
```diff
@@ -48,6 +48,14 @@ public function upgrade(Migration $migration) {
...
       $this->addFormVisibility();
       $this->addRequestSourceSetting();
+      $this->addEntityOption();
+   }
+
+   public function addEntityOption() {
+      $table = 'glpi_plugin_formcreator_entityconfigs';
+
+      $this->migration->addField($table, 'is_search_issue_visible', 'integer', ['after' => 'is_header_visible', 'value' => '-2']);
+      $this->migration->addPostQuery("UPDATE `glpi_plugin_formcreator_entityconfigs` SET `is_search_issue_visible`= 1 WHERE `entities_id` = 0");
```
Comment (btry) [inline install/upgrade_to_2.13.php:58]: these 2 lines must run if the field does not already exists. Currently, running the upgrade twice (after a failed attempt or a forced upgrade) may overwrite the setting for the root entity.

**Review by btry — CHANGES_REQUESTED** [review]

**Pull request merged**